### PR TITLE
[SASS-518] giftAid - overseas charity names

### PR DESCRIPTION
**Pull request by juamalBlackman**
### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-518](https://jira.tools.tax.service.gov.uk/browse/SASS-518)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you addressed warnings where appropriate?

##### After PRs been Reviewed
- [ ]  Have you checked the PR Builder passes?
- [ ]  Have you checked for merge conflicts?
- [ ]  Have you checked code coverage isn’t lower than previously?

### CODE REVIEW

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]
Please see above. Is there a reason why there are no view specs for this page?

**Comment by robert-webb-hmrc** [comment]
(also requires a rebase)

File: app/views/charity/GiftAidOverseasNameView.scala.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+        messages("charity.gift-aid-overseas-name.title.individual")
+    }
+}
+
+@headingForUse = @{(titleText)}
+
+@hint = {
+    <p>
+    @messages("charity.gift-aid-overseas-name.hint")
```
Comment (robert-webb-hmrc) [inline app/views/charity/GiftAidOverseasNameView.scala.html:45]: This should also not be a hint, but its own element. Please see screenshot of giftaid presentation 
Screenshot 2021-04-27 at 14 23 42

File: app/forms/validation/StringConstraints.scala [hunk 2]
```diff
@@ -27,7 +27,15 @@ object StringConstraints {
   val monetaryRegex = """\d+|\d*\.\d{1,2}"""
 
   val validateChar: String => Constraint[String] = msgKey => constraint[String](
-  x => if (x.matches(charRegex)) Valid else Invalid(msgKey)
+    x => if (x.matches(charRegex)) Valid else Invalid(msgKey)
```
[the remark this answers is not in the document]
Comment (juamalBlackman) [reply app/forms/validation/StringConstraints.scala:30]: awaiting confirmation on regex

[the remark this answers is not in the document]
Comment (juamalBlackman) [reply app/forms/validation/StringConstraints.scala:30]: updated to only include chars in error message

File: app/views/charity/GiftAidOverseasNameView.scala.html [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        messages("charity.gift-aid-overseas-name.title.individual")
+    }
+}
+
+@headingForUse = @{(titleText)}
+
+@content = {
+    <p>
+    @messages("charity.gift-aid-overseas-name.hint")
+    </p>
```
Comment (robert-webb-hmrc) [inline app/views/charity/GiftAidOverseasNameView.scala.html:46]: Please can we check the formatting (in the view file) and the spacing (in the actual page) here

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]
Can we please rebase this as well

File: app/forms/validation/StringConstraints.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (juamalBlackman) [reply app/forms/validation/StringConstraints.scala:30]: is only used here

File: app/forms/validation/StringConstraints.scala [hunk 4]
```diff
@@ -27,7 +27,15 @@ object StringConstraints {
...
+  )
+
+  def validateSize(maxChars: Int): String => Constraint[String] = msgKey => constraint[String](
+    x => if (x.length <= maxChars) Valid else Invalid(msgKey)
+  )
+
+  def validateNotDuplicate(maxChars: List[String]): String => Constraint[String] = msgKey => constraint[String](
```
[the remark this answers is not in the document]
Comment (juamalBlackman) [reply app/forms/validation/StringConstraints.scala]: changed

File: app/views/charity/GiftAidOverseasNameView.scala.html (hunk 1 above)
Comment (juamalBlackman) [reply app/views/charity/GiftAidOverseasNameView.scala.html]: removed extra class for bottom margin space

**Review by robert-webb-hmrc — APPROVED** [review]

**Pull request merged**